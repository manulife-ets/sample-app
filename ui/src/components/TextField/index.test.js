import React from 'react'
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import { TextField } from '.';

describe('Textfield', () => {
  test('it works', () => {
    const { container } = render(<TextField />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('font-family', "'Manulife JH Sans',sans-serif");
    
  })

  test('it renders an error', async () => {
    const { findByRole } = render(<TextField error='BOOM' />);
    const alert = await findByRole('alert');
    expect(alert).toHaveTextContent(/BOOM/i)
  })
})