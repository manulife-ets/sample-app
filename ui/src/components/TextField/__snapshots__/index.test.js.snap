// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textfield it works 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-family: 'Manulife JH Sans',sans-serif;
  position: relative;
  width: 100%;
}

.c1 {
  color: #8e90a2;
  cursor: text;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 0.875rem;
  padding: 0;
  position: relative;
}

.c2 {
  background: none;
  border: none;
  box-shadow: 0 1px 0 0 #8e90a2;
  color: #282b3e;
  font-size: 1rem;
  line-height: 1.875rem;
  outline: none;
  padding: 0.3125rem 0;
  -webkit-transition: box-shadow 240ms ease-in-out;
  transition: box-shadow 240ms ease-in-out;
  font-family: 'Manulife JH Sans',sans-serif;
}

.c2:focus {
  box-shadow: 0 2px 0 0 #00a758;
}

@media only screen and (max-width :414px) {
  .c0 {
    margin-bottom: 1rem;
  }
}

<div
  class="c0"
>
  <label
    class="c1"
  />
  <input
    class="c2"
    type="text"
    value=""
  />
</div>
`;
